File: src/imagebuilder/flyingtux/app/toolbase.py
from metux.util.specobject import SpecObject
from metux.util import log

class ToolBase(SpecObject):
    def __init__(self, spec, name):
        SpecObject.__init__(self, spec)
        self.my_toolname = name

    def info(self, text):
        log.info(self.my_toolname+': '+text)

    def abort(self, text):
        raise Exception(self.my_toolname+': '+text)

File: internal/app/toolbase.go
// Package app is the flyingtux orchestration layer: the Target (top-level
// config) and the Builder/Deploy/Runner pipeline tools (ports of
// flyingtux.target and flyingtux.app.*).
package app

import (
	"github.com/metux/flyingtux/internal/log"
	"github.com/metux/flyingtux/internal/spec"
)

// toolBase is the shared base for the pipeline tools (port of ToolBase).
type toolBase struct {
	*spec.SpecObj
	toolname string
	target   *Target
}

func (tb *toolBase) info(text string) { log.Info(tb.toolname + ": " + text) }
